rpcs3/rpcs3qt/gs_frame.cpp

```diff
@@ -456,15 +456,21 @@ bool gs_frame::event(QEvent* ev)
 				toggle_fullscreen();
 			}
 
-			int result;
+			int result = QMessageBox::Yes;
+			atomic_t<bool> called = false;
 
-			Emu.CallAfter([this, &result]()
+			Emu.CallAfter([this, &result, &called]()
 			{
 				m_gui_settings->ShowConfirmationBox(tr("Exit Game?"),
 					tr("Do you really want to exit the game?\n\nAny unsaved progress will be lost!\n"),
 					gui::ib_confirm_exit, &result, nullptr);
+
+				called = true;
+				called.notify_one();
```

Comment:
I never had trouble here. Why is it needed

Reply:
Thread-safety? unless the callback is executed from the same thread as the one called by Emu.CallAfter.

Reply:
I.e. wait for the thread to actually execute the callback before readig the result.